File: sbr-backend/models/Referral.js
const mongoose = require('mongoose');

const referralSchema = new mongoose.Schema({
  referrerId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  referralCode: {
    type: String,
    required: true
  },
  refereeName: {
    type: String,
    required: true,
    trim: true
  },
  refereePhone: {
    type: String,
    required: true,
    trim: true
  },
  productId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Product'
  },
  productName: {
    type: String,
    required: true,
    trim: true
  },
  status: {
    type: String,
    enum: ['Pending', 'Contacted', 'Purchased', 'Reward Credited', 'Rejected'],
    default: 'Pending'
  },
  purchaseAmount: {
    type: Number,
    default: 0
  },
  rewardAmount: {
    type: Number,
    default: 0
  },
  notes: {
    type: String,
    default: ''
  }
}, { timestamps: true });

module.exports = mongoose.model('Referral', referralSchema);

File: sbr android/app/src/main/java/com/sbr/sms/data/models/Referral.kt
package com.sbr.sms.data.models

import com.google.gson.annotations.SerializedName

data class Referral(
    @SerializedName("id", alternate = ["_id"]) val id: String = "",
    val referrerId: Any? = null,
    val referralCode: String = "",
    val refereeName: String = "",
    val refereePhone: String = "",
    val productId: String? = null,
    val productName: String = "",
    val rewardAmount: Double = 0.0,
    val notes: String? = null,
    val status: String = "Pending",
    val purchaseAmount: Double? = null,
    val createdAt: String? = null
)

data class ReferralClaim(
    @SerializedName("id", alternate = ["_id"]) val id: String = "",
    val userId: Any? = null,
    val userName: String = "",
    val userPhone: String = "",
    val amount: Double = 0.0,
    val payoutMethod: String = "", // UPI, BANK_TRANSFER, CASH_STORE
    val payoutDetails: String = "",
    val status: String = "Pending", // Pending, Approved, Rejected, Completed
    val transactionRef: String? = null,
    val adminNotes: String? = null,
    val createdAt: String? = null
)

data class ReferralDashboard(
    val referralCode: String = "",
    val totalInvited: Int = 0,
    val convertedCount: Int = 0,
    val totalEarnings: Double = 0.0,
    val claimedEarnings: Double = 0.0,
    val availableBalance: Double = 0.0,
    val pendingEarnings: Double = 0.0,
    val referrals: List<Referral> = emptyList(),
    val claims: List<ReferralClaim> = emptyList()
)
